### ISSUE TITLE
Reduce outdated gdb and valgrind packages to dummy script packages

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/sage#30158]
<div id="comment:0"></div>

These are devtools, not needed for building.

Last update to `gdb` was made in 2018

Last update to `valgrind` was made in 2019

CC:  @dimpase @jhpalmieri @orlitzky

Component: **packages: experimental**

_Issue created by migration from https://trac.sagemath.org/ticket/30158_

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -4,3 +4,5 @@
 https://github.com/mkoeppe/sage/runs/876312037
 
 
+https://ftp.gnu.org/gnu/gdb/
+
``````

**Comment by mkoeppe** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,8 +1,6 @@
-We should ship a version that matches the shipped gcc package, for use in portability testing on system that we support.
+These are devtools, not needed for building.
 
-Currently we have gdb 8.2, and installation fails at least on `ubuntu-xenial`... perhaps only because of missing `makeinfo`?
-https://github.com/mkoeppe/sage/runs/876312037
+Last update to `gdb` was made in 2018
 
+Last update to `valgrind` was made in 2019
 
-https://ftp.gnu.org/gnu/gdb/
-
``````

**Comment by mkoeppe** [comment]
> Replying to [@jhpalmieri](#comment%3A12):
> 
> > If someone uses a system Python without `.supp` files, can they use Sage-distributed `.supp` files instead? For what it's worth, homebrew doesn't seem to provide these
> 
> Yes, the idea of the `sage-valgrind` script is that it provides `--suppressions` options to valgrind.
> 
> I like the idea of this script, but the details need updating. I needed valgrind only once in the last 5 years and noticed that it was outdated but didn't have the energy to make the proper update. For such rare uses, an updated script (and a vendored copy of the Python suppressions -- note that they are NOT frequently updated upstream) would be of great value.

The scripts are updated in:
- #37569

The packages are reduced to dummy in:
- #38306
